File: odttf2ttf/ODTTF.cs
using System;
using System.IO;

public class ODTTF
{
	public static byte[] Deobfuscate(string odttfFilePath)
	{
		byte[] fontData;

		using (FileStream fs = new FileStream(odttfFilePath, FileMode.Open))
		{
			fontData = new byte[fs.Length];
			fs.Read(fontData, 0, fontData.Length);
		}

		string fileNameWithoutExtension = Path.GetFileNameWithoutExtension(odttfFilePath);

		string guid = fileNameWithoutExtension.Replace("-", "");

		byte[] key = new byte[16];
		for (int i = 0; i < key.Length; i++)
		{
			key[key.Length - 1 - i] = Convert.ToByte(guid.Substring(i * 2, 2), 16);
		}

		for (int i = 0; i < 32; i++)
		{
			fontData[i] ^= key[i % key.Length];
		}

		return fontData;
	}
}

File: js/odttf.js
function ODTTF(){
    this.deobfuscate = function (fontData, guid) {
        guid = guid.replace(/-/g, "");

        let key = [];
        const keyLength = 16;

        for (let i = 0; i < keyLength; i++)
            key[keyLength - 1 - i] = parseInt(guid.substr(i * 2, 2), 16);

        for (let i = 0; i < 32; i++)
            fontData[i] ^= key[i % key.length];
    };
}